core/config/schwarz_config.cpp

```diff
@@ -0,0 +1,52 @@
+// SPDX-FileCopyrightText: 2017 - 2024 The Ginkgo authors
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <ginkgo/core/config/config.hpp>
+#include <ginkgo/core/config/registry.hpp>
+#include <ginkgo/core/config/type_descriptor.hpp>
+#include <ginkgo/core/distributed/preconditioner/schwarz.hpp>
+
+#include "core/config/config_helper.hpp"
+#include "core/config/dispatch.hpp"
+#include "core/config/type_descriptor_helper.hpp"
+
+
+namespace gko {
+namespace config {
+
+
+template <>
+deferred_factory_parameter<gko::LinOpFactory> parse<LinOpFactoryType::Schwarz>(
+    const pnode& config, const registry& context, const type_descriptor& td)
+{
+    auto updated = update_type(config, td);
+    // We can not directly dispatch the global index type without consider local
+    // index type, which leads the invalid index type <int64, int32> in
+    // compile time.
```

Comment:
This logic would be necessary for all distributed types with Global and Local Index types, right ?

Reply:
yes, but no other cases require it currently.

Reply:
we limit linopfactory only